kbase-extension/static/kbase/js/widgets/kbaseTabs.js

```diff
@@ -241,47 +282,72 @@ define(['kbwidget', 'bootstrap', 'jquery', 'kbaseDeletePrompt'], (
             return this.data('tabs')[tabName];
         },
 
-        showTab: function (tab) {
-            this.data('nav')[tab].find('a').trigger('click');
+        showTab: function (tabName) {
+            this.data('nav')[tabName].find('a').trigger('click');
         },
 
         removeTab: function (tabName) {
             const $tab = this.data('tabs')[tabName];
             const $nav = this.data('nav')[tabName];
+            const deleteNav = () => {
+                $nav.remove();
+                delete this.data('nav')[tabName];
+            };
+
+            const deletePanel = () => {
+                $tab.remove();
+                delete this.data('tabs')[tabName];
+            };
+
+            const deleteTab = () => {
+                deleteNav();
+                deletePanel();
+            };
 
             if ($nav.hasClass('active')) {
-                if ($nav.next('li').length) {
-                    $nav.next().find('a').trigger('click');
+                const [$nextTab, $nextPanel] = (() => {
+                    if ($nav.next('li').length) {
+                        return [$nav.next('li'), $tab.next('[role="tabpanel"]')];
+                    } else if ($nav.prev('li').length) {
+                        return [$nav.prev('li'), $tab.prev('[role="tabpanel"]')];
+                    } else {
+                        return [null, null];
+                    }
+                })();
+
+                if ($nextTab) {
+                    // We remove the tab nav button, but hide the tab
+                    // panel - this causes it to take up the same space and become
+                    // transparent. This helps prevent the tab shrinking then expanding.
+                    deleteNav();
+                    $tab.css('visibility', 'hidden');
+
+                    // We delete the panel after the new one is shown.
+                    $nextPanel.one('shown', () => {
+                        deletePanel();
+                    });
+
+                    // Simulate a user click of the next tab to display.
+                    $nextTab.find('a').trigger('click');
                 } else {
-                    $nav.prev('li').find('a').trigger('click');
+                    deleteTab();
                 }
+            } else {
+                deleteTab();
             }
-
-            $tab.remove();
-            $nav.remove();
-
-            this.data('tabs')[tabName] = undefined;
-            this.data('nav')[tabName] = undefined;
         },
 
         deletePrompt: function (tabName) {
```

Comment:
simplified as there was no need to have an anonymous callback function call a method.